pkg/sosreport/sosreport.jsx

```diff
@@ -89,7 +89,7 @@ function sosLister() {
     }
 
     function update_reports() {
-        cockpit.script('find /var/tmp -maxdepth 1 -name \'*sosreport-*.tar.*\' -print0 | xargs -0 -r stat --printf="%n\\r%W\\n"', { superuser: true, err: "message" })
+        cockpit.script('printf \"\$(stat -c \'%n\\r%Y\\n\' /var/tmp/*sosreport-*.tar.*)\"', { superuser: true, err: "message" })
```

Comment:
Changing the `%W` to a `%Y` is fine, but what does the `printf` wrapper do? AFAICS that plus the `$()` really shouldn't be here.

Also, sosreport file names can contain user-provided labels, which may well include spaces (breaking the glob) and other shell variables.

I think we should finish #19293 first, then we can do exactly this kind of operation in a robust and also much simpler manner, without any shell stuff.

Reply:
`printf` + `$()` is a quite ugly hook to format output exactly how it was by `--printf` arg of xargs.  (i.e. '\r' and '\n' need to be interpreted).

To be honest, I think a better way would be to change the (lines, path and date) parsing instead twisting shell command to output the "easiest" output.

Finally, you're right: using a proper method to list files and attributes will definitely solve this kind of "shell glue" issue.

Reply:
Note that `--printf` is an option of `stat`, not of `xargs`. That's why I thought just calling `stat --printf.. /var/*` should be enough. I was actually wrong about breaking the globbing -- files with spaces should be handled just fine.

But yes, a proper stat() fslist API is still much nicer, of course, but we don't have it just yet.